app/controllers/users_controller.rb

```diff
@@ -61,11 +61,15 @@ def admin_redirect
   end
 
   def user_valid_destroy
-    @user.reviews.delete_all
+    @user.reviews.each do |review|
```

Comment:
I don't think you really have to, but what's going on is actually really interesting: 

You are defining (through very non-obvious symbols) that you are creating a block-symbol: `&:delete`. This represents a block that then sends a message to the method that called it. 

You then pass that block to `each` (same as you do when using `do...end` blocks) which sends that message *in turn* to each item that you are iterating over. 

So, this: 

`@user.reviews.each(&:delete)`

is functionally the same as: 
```
@user.reviews.each do |r| 
  r.delete
end
```